crates/neon-runtime/src/napi/primitive.rs

```diff
@@ -43,3 +43,9 @@ pub unsafe fn number_value(env: Env, p: Local) -> f64 {
     );
     value
 }
+
+/// Mutates the `out` argument provided to refer to a newly created `Local` containing a
+/// JavaScript symbol.
+pub unsafe fn symbol(out: &mut Local, env: Env, desc: Local) {
+    napi::create_symbol(env, desc, out as *mut Local);
+}
```

Comment:
This method needs to validate `napi_status`.

```suggestion
pub unsafe fn symbol(env: Env, desc: Local) -> Local {
    let local = MaybeUninit::uninit();
    
    assert_eq!(napi::create_symbol(env, desc, local.as_mut_ptr()), napi::Status::Ok);
    
    local.assume_init();
}
```

Reply:
For this part I was definitely playing the follow-the-pattern game. Is there a reason why the other create primitives aren't validated? I can understand `null`/`undefined`/`boolean` as they are effectively just getters for global values, but `number` doesn't have any validation.

Reply:
JsNumber should be validating it since it's effectively infallible due to protections on the Rust side.

Thinking about this one a bit more, it should probably return something to indicate an error since there are valid cases where it could `throw`.

Maybe check if it's throwing and return None and _then_ assert that it's ok.

Reply:
I'm not following. I understand this part:
> JsNumber should be validating it since it's effectively infallible due to protections on the Rust side.

In `neon_runtime::primitive::number()`, the value passed in has to be an `f64`, so it's inherently validated by the Rust type system.

While in `neon_runtime::primitive::symbol()`, `desc` has to be either a null pointer or a `Local` representing a `ValueType::String`. Adding the status check makes sense to me given that someone could pass in a different `Local` type, but I don't understand this part:
> Thinking about this one a bit more, it should probably return something to indicate an error since there are valid cases where it could throw.

What does this have to do with `throw`? If we're in a throwing state, and this function returns `Option<Local>` We'd have to expect/unwrap it in `JsSymbol::new_internal()` in order to return `Handle<JsSymbol>`. If we then unwrap the `None`, that panic would override the current throw state which seems to defeat the point. 

If it's the case that you're mistaking this function for the description getter, that one currently returns `None` when in a throwing state:
```rust
    let sym = cx.symbol("description");
    let opt1 = sym.description(&mut cx);
    assert!(opt1.is_some());
    cx.throw_type_error::<_, ()>("entering throw state with TypeError").unwrap_err();
    let opt2 = sym.description(&mut cx);
    assert!(opt2.is_none());
```
Although this is a side effect of `napi::get_property()` failing in `neon_runtime::object::get_string()` rather than explicitly checking for the throw state in `JsSymbol::description()`.

Reply:
You're right! I was thinking this method accepts a `str` as a description and converting to `JsString` could fail, but it's already a `napi_value` here. 🤦 

I agree, checking the status shouldn't be necessary. It might not hurt to check in a `debug_assert`.